lib/runtime/wasmer/imports.go

```diff
@@ -1060,6 +1063,33 @@ func ext_default_child_storage_clear_prefix_version_1(context unsafe.Pointer, ch
 	}
 }
 
+//export ext_default_child_storage_clear_prefix_version_2
+func ext_default_child_storage_clear_prefix_version_2(context unsafe.Pointer, childStorageKey, prefixSpan,
+	limitSpan C.int64_t) C.int64_t {
+	logger.Debug("executing...")
+
+	instanceContext := wasm.IntoInstanceContext(context)
+	ctx := instanceContext.Data().(*runtime.Context)
+	storage := ctx.Storage
+
+	keyToChild := asMemorySlice(instanceContext, childStorageKey)
+	prefix := asMemorySlice(instanceContext, prefixSpan)
+
+	var limit *uint32
+	err := scale.Unmarshal(asMemorySlice(instanceContext, limitSpan), limit)
+	if err != nil {
+		logger.Errorf("failed to decode limit: %s", err)
+	}
+
+	err = storage.ClearPrefixInChildWithLimit(keyToChild, prefix, *limit)
+	if err != nil {
+		logger.Errorf("failed to clear prefix in child with limit: %s", err)
+	}
+
+	// TODO: Should this always be 0 or could this be something else as well?
+	return C.int64_t(0)
```

Comment:
Following the spec the returned value should be a pointer-size to the following variant:

- Variant 0 indicates that all keys of the child storage have been removed, followed by the number of removed keys
- Variant 1 indicates that there are remaining keys, followed by the number of removed keys.

ref: https://spec.polkadot.network/chap-host-api#id-version-2-prototype-3

You can see how they implemented it in substrate:

```rs
#[derive(PassByCodec, Encode, Decode)]
pub enum KillStorageResult {
	/// All keys to remove were removed, return number of iterations performed during the
	/// operation.
	AllRemoved(u32),
	/// Not all key to remove were removed, return number of iterations performed during the
	/// operation.
	SomeRemaining(u32),
}

#[version(2)]
fn clear_prefix(&mut self, prefix: &[u8], limit: Option<u32>) -> KillStorageResult
```
- clear prefix v2: https://github.com/paritytech/substrate/blob/d58481569817f4a2074dbab014fa629dea7e0ec9/primitives/io/src/lib.rs#L186
- the returned variant: https://github.com/paritytech/substrate/blob/d58481569817f4a2074dbab014fa629dea7e0ec9/primitives/io/src/lib.rs#L96